src/transformations/correction/geos.jl

```diff
@@ -0,0 +1,22 @@
+#=
+# LibGEOS correction
+
+This file defines exportable skeleton code for a "GEOSCorrection" which 
+simply invokes `LibGEOS.makeValid` on the geometry.
+
+The glue code is defined in the extension on LibGEOS.
+
+=#
+
+
+"""
+    GEOSCorrection() <: GeometryCorrection
+
+This correction runs `LibGEOS.makeValid` on the highest level of each geometry available.
+
+See also [`GeometryCorrection`](@ref).
+"""
+struct GEOSCorrection <: GeometryCorrection end
+
+application_level(::GEOSCorrection) = TraitTarget(GI.MultiPolygonTrait(), GI.PolygonTrait(), GI.LinearRingTrait(), GI.MultiLineStringTrait(), GI.LineStringTrait(), GI.MultiPointTrait(), GI.PointTrait())
+
```

Comment:
should we add in the actual `makeValid` call? Or is that in a different PR?

Reply:
I guess I also don't really understand why this wouldn't be in the `LibGEOS` extensions folder since it is going to call `LibGEOS` code according to the comments and I would hope that one day we would have our own code for this.

Reply:
We cant add new methods or types in the extensions folder, they wont be visible to users. I think we need this file just for the `GEOScorrection` struct.

Reply:
This is a skeleton - I was going to add the actual correction in the extension, but the type has to be (a) <: AbstractGeometryCorrection and (b) accessible to users, which is not possible to do if it's implemented in the extension. 

The issue I ran into is that LG can return a multipolygon given polygon input, so I might have to reimplement apply for this with some hooks in the reconstructor...

Reply:
Ah so the other ones work because they are also names of functions we have. Which is why we need the [not_implemented_yet.jl](https://github.com/JuliaGeo/GeometryOps.jl/pull/100/files#diff-14f503d5c8d2ca266096779fcd5ad1d48dd2667e71a52e0bce8cc12f9e85dcd6) file? Is that an accurate understanding?

Reply:
Yep, that's why `buffer` works for example

Reply:
Also I just want to make sure that this one does return `GI.Polygon`. I feel like it would be confusing since it isn't one that we are passing `GEOS()` into, which would at least clearly mark that it would return an `LG` polygon.

Reply:
GEOSCorrection has been removed for now to not complicate this PR.